rclcpp/src/rclcpp/node_options.cpp

```diff
@@ -93,11 +93,20 @@ NodeOptions::get_rcl_node_options() const
     int c_argc = 0;
     std::unique_ptr<const char *[]> c_argv;
     if (!this->arguments_.empty()) {
-      c_argc = static_cast<int>(this->arguments_.size()) + 1;
+      auto it = std::find(
+        this->arguments_.cbegin(), this->arguments_.cend(), RCL_ROS_ARGS_FLAG);
+
+      c_argc = static_cast<int>(this->arguments_.size());
+      if (it == this->arguments_.cend()) c_argc += 1;
```

Comment:
nitpick: Please always use `{}` for if statements.

Reply:
See 8772bd0d73d2e9bf958e7f545bf3358ba226f61f.

Reply:
Thanks!